src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Downloads imagery datasets e.g. Landsat, Sentinel, and MODIS
+#              using EODAG API.
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+
+# %Module
+# % description: Eodag interface to install imagery datasets from various providers.
+# % keyword: imagery
+# % keyword: eodag
+# % keyword: sentinel
+# % keyword: landsat
+# % keyword: modis
+# % keyword: datasets
+# % keyword: download
+# %end
+
+# %option
+# % key: dataset
+# % type: string
+# % description: Imagery dataset to search for
+# % required: yes
+# % answer: S1_SAR_GRD
+# % guisection: Filter
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output
+# % description: Name for output directory where to store downloaded data OR search results
+# % required: no
+# % guisection: Output
+# %end
+
+# %option G_OPT_M_DIR
+# % key: format
+# % type: string
+# % description: Don't download and show/save search results as: plain;Plain text output;json;JSON
+# % required: no
+# %end
```

Comment:
```suggestion
# %option
# % key: format
# % type: string
# % description: Output format
# % required: no
# % options: plain,json
# % answer: json
# %end
```
The appropriate standard parser option is [G_OPT_F_FORMAT](https://grass.osgeo.org/grass84/manuals/parser_standard_options.html). Please note that it is quite new in GRASS GIS 8.4 (actually added by another GSoC student: https://github.com/OSGeo/grass/pull/3704/files). So for the time beeing, (and to support GRASS GIS < 8.4), you may use the above suggestion / work-around...

For printing you probably have to add another `print` option... (as format only defines the format for printing (or output to file)...